### ISSUE TITLE
Implement "cleanup" method

### ISSUE THREAD

**Issue by leotrs** [ComplexGroupInteractions/xgi#36]
~Make sure the documentation highlights the fact that the user should remember to call `remove_isolates` and other related functions.~
Implement a `cleanup` method that performs various common clean up tasks, see comment down thread.

**Comment by leotrs** [comment]
#39 implements `remove_isolates`

**Comment by leotrs** [comment]
Other ideas for 2 `cleanup` method: remove isolates, extract largest component, remove duplicate/multi edges, relabel the node ids.

**Comment by nwlandry** [comment]
So you're saying that we have a cleanup method that does these four steps? I like that idea and also we should add the possibility to collapse loopy hyperedges.

**Comment by leotrs** [comment]
> So you're saying that we have a cleanup method that does these four steps?

We don't have it yet, but I want to implement it.

> I like that idea and also we should add the possibility to collapse loopy hyperedges.

Good catch!

**Comment by leotrs** [comment]
For now I think it'd be nice to have something like the following:
```
class Hypergraph:
    ...
    def cleanup(self):
        self.remove_isolates()
        self.remove_singletons()
        self.collapse_loopy_edges()    # not implemented yet
        self.remove_duplicates()     # not implemented yet
        self.relabel_nodes()     # not implemented yet
    ...
```

What other operations should `cleanup` perform?

**Comment by leotrs** [comment]
Closed by #161
